src/20240421_Multi_AprilFinanceUpdate/AaveV2Polygon_AprilFinanceUpdate_20240421.sol

```diff
@@ -0,0 +1,181 @@
+      amount,
+      address(AaveV3Polygon.COLLECTOR),
+      0
+    );
+  }
+
+  function _withdrawMatic() internal {
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV2PolygonAssets.WMATIC_A_TOKEN,
+      address(this),
+      IERC20(AaveV2PolygonAssets.WMATIC_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) -
+        1 ether
+    );
+
+    AaveV2Polygon.POOL.withdraw(
+      AaveV2PolygonAssets.WMATIC_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    AaveV3Polygon.COLLECTOR.transfer(
+      AaveV3PolygonAssets.WMATIC_A_TOKEN,
+      address(this),
+      IERC20(AaveV3PolygonAssets.WMATIC_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) -
+        1 ether
+    );
+
+    AaveV3Polygon.POOL.withdraw(
+      AaveV3PolygonAssets.WMATIC_UNDERLYING,
+      type(uint256).max,
+      address(this)
+    );
+
+    IWMatic(AaveV3PolygonAssets.WMATIC_UNDERLYING).withdraw(
+      IERC20(AaveV3PolygonAssets.WMATIC_UNDERLYING).balanceOf(address(this))
+    );
+
+    (bool ok, ) = address(plasmaBridge).call{value: address(this).balance}('');
+
+    if (!ok) revert FailedToSendMATIC();
+  }
+
+  /**
+   * @notice Withdraws v2 tokens to bridge
+   */
+  function _withdrawV2Tokens() internal {
+    TokenToBridge[] memory tokens = new TokenToBridge[](2);
+    tokens[0] = TokenToBridge(
+      AaveV2PolygonAssets.DAI_UNDERLYING,
+      AaveV2PolygonAssets.DAI_A_TOKEN,
+      IERC20(AaveV2PolygonAssets.DAI_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 1 ether
+    );
+    tokens[1] = TokenToBridge(
+      AaveV2PolygonAssets.USDC_UNDERLYING,
+      AaveV2PolygonAssets.USDC_A_TOKEN,
+      IERC20(AaveV2PolygonAssets.USDC_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 1e6
+    );
+
+    uint256 tokensLength = tokens.length;
+    for (uint256 i = 0; i < tokensLength; i++) {
+      AaveV3Polygon.COLLECTOR.transfer(tokens[i].aToken, address(this), tokens[i].balance);
+
+      AaveV2Polygon.POOL.withdraw(tokens[i].underlying, type(uint256).max, address(bridge));
+    }
+  }
+
+  /**
+   * @notice Withdraws v3 tokens to bridge
+   */
+  function _withdrawV3Tokens() internal {
+    TokenToBridge[] memory tokens = new TokenToBridge[](2);
+    tokens[0] = TokenToBridge(
+      AaveV3PolygonAssets.DAI_UNDERLYING,
+      AaveV3PolygonAssets.DAI_A_TOKEN,
+      IERC20(AaveV3PolygonAssets.DAI_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 1 ether
+    );
+    tokens[1] = TokenToBridge(
+      AaveV3PolygonAssets.USDC_UNDERLYING,
+      AaveV3PolygonAssets.USDC_A_TOKEN,
+      IERC20(AaveV3PolygonAssets.USDC_A_TOKEN).balanceOf(address(AaveV3Polygon.COLLECTOR)) - 1e6
+    );
+
+    uint256 tokensLength = tokens.length;
+    for (uint256 i = 0; i < tokensLength; i++) {
+      AaveV3Polygon.COLLECTOR.transfer(tokens[i].aToken, address(this), tokens[i].balance);
+
+      AaveV3Polygon.POOL.withdraw(tokens[i].underlying, type(uint256).max, address(bridge));
+    }
+  }
+
+  receive() external payable {}
```

Comment:
```suggestion
```
@efecarranza This is not needed as `address(this)` is the executor in this case which will receive the MATIC token.